modules/xfem/src/base/XFEM.C

```diff
@@ -59,7 +59,7 @@ XFEM::getCrackTipOrigin(std::map<unsigned int, const Elem* > & elem_id_crack_tip
   std::map<unsigned int, const Elem*> elem_id_map;
 
   int m = -1;
-  for (mit1 = _elem_crack_origin_direction_map.begin(); mit1 != _elem_crack_origin_direction_map.end(); mit1++)
+  for (; mit1 != _elem_crack_origin_direction_map.end(); ++mit1)
```

Comment:
I changed the iterators here to be declared in the scope of the loop.  This code still needs some work, but fixing it is outside the scope of this issue.